script/test_fixtures.sh

```diff
@@ -1,42 +1,61 @@
 #!/bin/bash
-set -ex
-
-make
+set -e
 
 test_folder() {
-    find "$1" -name "*_expected.rb" -maxdepth 1 | while read -r file
-    do
-        # shellcheck disable=SC2001
-        time ruby --disable=gems rubyfmt.rb "$(echo "$file" | sed s/expected/actual/)" > /tmp/out.rb
+    current_dir="$1"
+
+    ## Check that both files or neither exist
+    if [ -f "$current_dir/expected.rb" ] || [ -f "$current_dir/actual.rb" ]; then
+      [ -f "$current_dir/expected.rb" ] || (echo "$current_dir is missing the expected.rb file" && exit 1)
+      [ -f "$current_dir/actual.rb" ] || (echo "$current_dir is missing the actual.rb file" && exit 1)
+
+      ## Test if the formatting works as expected
+      time ruby --disable=gems rubyfmt.rb "$current_dir/actual.rb" > /tmp/out.rb
+      if ! diff -u /tmp/out.rb "$current_dir/expected.rb" > /tmp/diff.out
+      then
+        echo "got diff between formated formatted actual and expected"
+        cat /tmp/diff.out
+        exit 1
+      fi
+
+      ## Test if the formatting is idempotent
+      time ruby --disable=gems rubyfmt.rb "$current_dir/expected.rb" > /tmp/out.rb
+      if ! diff -u /tmp/out.rb "$current_dir/expected.rb" > /tmp/diff.out
+      then
+        echo "got diff between formatted expected and expected (not idempotent)"
+        cat /tmp/diff.out
```

Comment:
I also captured the diff output and only output on a failure to keep less in the terminal. The rust still logs a lot and I didn't capture that so this could definitely be removed if wanted